core/src/main/java/org/apache/calcite/rel/rules/JoinCommuteRule.java

```diff
@@ -145,6 +145,12 @@ public static RelNode swap(Join join, boolean swapOuterJoins,
         .build();
   }
 
+  public boolean matches(RelOptRuleCall call) {
+    Join join = call.rel(0);
+    // SEMI and ANTI join cannot be swapped.
+    return join.getJoinType().projectsRight();
+  }
```

Comment:
LOL. 😝
I remembered I created that JIRA but not immediately create a pull request, because I had to come up with a test case.
The code change looks good to me, but I think we'd better add a test case. :)

Reply:
You're right. I will create a test case.